Code/Common/include/sitkImage.h

```diff
@@ -371,6 +371,38 @@ namespace simple
 
     std::string ToString( ) const;
 
+    /** \brief Convert the a first dimension to the components for image with vector pixel type.
+     *
+     * This method will convert a scalar image to a vector image with
+     * the number of components equal to the size of the first
+     * dimension. If the image is already a vector image then the
+     * image is returned.
```

Comment:
Should it also say something about loosing the metadata information for the first axis? What I mean is that the operation is lossy with respect to metadata: `scalarImage.ToVectorImage().ToScalarImage() + scalarImage` will throw an exception because the metadata is different.

Reply:
This is a more complicated issues than just documentation. The spacing and origin have a straight forward way to drop  a dimension and prepend a 1 or 0 respectfully when adding a dimensions. However the Direction matrix can have some degenerative cases when the extract sub-matrix is not appropriate.

The most similar thing that exists in ITK is the ExtractImageFilter, which has multiple direction collapse strategies and not working default.

Reply:
A possible punting approach, which I think is valid, is to limit these operations to images with identity direction cosine matrices. I suspect this will address all relevant use cases. 

Extend the capabilities of these functions to non identity cosine matrices only if the need arises. Add the functionality only when there is a concrete use case.

Reply:
>A possible punting approach, which I think is valid, is to limit these operations to images with identity direction cosine matrices. I suspect this will address all relevant use cases.

Perhaps just the "component" dimension needs to be identity. This would follow the intended functionality of converting to and from vector images.